maestro-cli/src/main/java/maestro/cli/report/HtmlTestSuiteReporter.kt

```diff
@@ -0,0 +1,108 @@
+import okio.Sink
+import okio.buffer
+import org.slf4j.LoggerFactory
+import java.io.BufferedReader
+import java.io.FileReader
+import maestro.cli.report.TestDebugReporter
+
+class HtmlTestSuiteReporter : TestSuiteReporter {
+    override fun report(summary: TestExecutionSummary, out: Sink) {
+        val bufferedOut = out.buffer()
+        val htmlContent = buildHtmlReport(summary)
+        bufferedOut.writeUtf8(htmlContent)
+        bufferedOut.close()
+    }
+
+    private fun getTestStep(): Array<Array<String>> {
+        val debugOutputPath = TestDebugReporter.getDebugOutputPath()
+        val filePathLog = "${debugOutputPath}/maestro.log"
+        var reader: BufferedReader? = null
+        var testStep = emptyArray<String>()
+        var testSteps = emptyArray<Array<String>>()
+    
+        try {
+            reader = BufferedReader(FileReader(filePathLog))
+            var line: String = ""
+    
+            while (reader.readLine().also { line = it } != null) {
+                // Process each line
+                if(line.contains("m.cli.runner.TestSuiteInteractor") && !line.contains("Run") && !line.contains("Define variables") && !line.contains("Apply configuration")){
+                    if(line.contains("COMPLETED") || line.contains("FAILED")){
+                        testStep += line.replace("[INFO ] m.cli.runner.TestSuiteInteractor ", "")
+                    }
+                }
+                if(line.contains("m.cli.runner.TestSuiteInteractor - Stop") && line.contains("COMPLETED")){
+                    testSteps += testStep
+                    testStep = emptyArray<String>()
+                }
+            }
+        } catch (e: Exception) {
+        } finally {
+            try {
+                reader?.close()
+            } catch (e: Exception) {
+                println("An error occurred while closing the file: ${e.message}")
+            }
+        }
+        return testSteps
+    }
+
+    private fun getFailedTest(summary: TestExecutionSummary): Array<String>{
+        var failedTest = emptyArray<String>()
+        for (suite in summary.suites) {
+            for(flow in suite.flows){
+                if(flow.status.toString() == "ERROR"){
+                    failedTest += flow.name
+                }
+            }
+        }
+        return failedTest
+    }
+
+    private fun buildHtmlReport(summary: TestExecutionSummary): String {
+    val htmlBuilder = StringBuilder()
+    val testSteps = getTestStep()
+    var idx = 0
+    var failedTest = getFailedTest(summary)
+
+    htmlBuilder.append("<!doctype html><html lang=\"en\"><head><meta charset=\"utf-8\"><meta name=\"viewport\" content=\"width=device-width, initial-scale=1\"><title>Maestro Test Report</title><link href=\"https://cdn.jsdelivr.net/npm/bootstrap@5.3.3/dist/css/bootstrap.min.css\" rel=\"stylesheet\" integrity=\"sha384-QWTKZyjpPEjISv5WaRU9OFeRpok6YctnYmDr5pNlyT2bRjXh0JMhjY6hW+ALEwIH\" crossorigin=\"anonymous\"></head><body>")
```

Comment:
I understand the desire to roll your own html report instead of going with say Allure or so which I'm generally ok with, but I'm curious whether you considered using any type of templating engine here instead of building the full report as a string?